torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -1027,6 +1027,10 @@ bool PropagateTensorShapeOnNode(Node * node, bool insert_expands) {
       if (auto shape = node->get<std::vector<int64_t>>(shape_input)) {
         return tensor_types.at(0)->withDim(shape->size());
       }
+      auto input_node = node->namedInput(shape_input)->node();
+      if (input_node->kind() == prim::ListConstruct) {
+        return tensor_types.at(0)->withDim(input_node->inputs().size());
+      }
```

Comment:
Can we make this a "figure out size of this list" helper, and use it for e.g. `aten::expand` too?

Reply:
As far as I can tell this function (reshape_prop) is already used for aten::expand, aten::reshape. In addition, reshape_prop is the "figure out size of this list" helper. I can rename it so that it is clearer